src/components/Popups/PopupBase.tsx

```diff
@@ -0,0 +1,88 @@
+import * as React from 'react';
+import classNames from 'classnames';
+import createPopper, { Instance, Options } from './Popper';
+import './PopupBase.scss';
+
+type Props = React.HTMLAttributes<HTMLDivElement> & {
+    children: React.ReactNode;
+    className?: string;
+    options: Partial<Options>;
+    reference: HTMLElement;
+};
+
+export default class PopupBase extends React.PureComponent<Props> {
+    static defaultProps = {
+        options: {},
+    };
+
+    popper?: Instance;
+
+    popupRef: React.RefObject<HTMLDivElement> = React.createRef();
+
+    componentDidMount(): void {
+        this.popper = this.createPopper();
+    }
+
+    componentDidUpdate(prevProps: Props): void {
+        const { options: prevOptions, reference: prevReference } = prevProps;
+        const { options, reference } = this.props;
+
+        if (!this.popper) {
+            return;
+        }
+
+        if (options !== prevOptions) {
+            this.popper.setOptions(options);
+        }
+
+        if (reference !== prevReference) {
+            this.popper.destroy();
+            this.popper = this.createPopper();
+        }
+    }
+
+    componentWillUnmount(): void {
+        if (this.popper) {
+            this.popper.destroy();
+        }
+    }
+
+    createPopper = (): Instance | undefined => {
+        const { options, reference } = this.props;
+        const { current: popupRef } = this.popupRef;
+
+        if (!popupRef) {
+            return undefined;
+        }
+
+        return createPopper(reference, popupRef, options);
+    };
+
+    handleEvent = (event: React.SyntheticEvent): void => {
+        event.preventDefault();
+        event.stopPropagation();
+        event.nativeEvent.stopImmediatePropagation();
+    };
+
+    render(): JSX.Element {
+        const { children, className, options, reference, ...rest } = this.props; // eslint-disable-line @typescript-eslint/no-unused-vars
```

Comment:
Given that there are only 4 props, would it be better to avoid `...rest` and just destructure `children` and `className` or should we import lodash/omit so that we can do `...rest`?

Reply:
Yeah, I had lodash/omit in here at one point, but it seemed like overkill. Our default eslint rules don't even complain about this pattern and it's super common, so it might be worth disabling this rule altogether..